File: notif_usb.c
#include <stdio.h>
#include <string.h>
#include <process.h>

int main(int argc, char *argv[]) {
    if (argc < 3) {
        fprintf(stderr, "Usage: notif_usb.exe <titre> <message>\n");
        return 1;
    }

    char *cmd[] = {
        "adb", "shell", "cmd", "notification", "post",
        "-S", "bigtext",
        "-t", argv[1],
        "python_notif",
        argv[2],
        NULL
    };

    _spawnvp(_P_WAIT, "adb", (const char * const *)cmd);

    return 0;
}

File: notif_usb.py
import subprocess
import sys
import shlex

def envoyer_notification(titre, message, tag="python_notif"):
    titre_echap = shlex.quote(titre)
    message_echap = shlex.quote(message)
    tag_echap = shlex.quote(tag)

    cmd = [
        "adb", "shell", "cmd", "notification", "post",
        "-S", "bigtext",
        "-t", titre_echap,
        tag_echap,
        message_echap,
    ]

    result = subprocess.run(cmd, capture_output=True, text=True)

def main():
    print("notification")
    titre = input("Titre: ").strip()
    message = input("Contenu: ").strip()

    envoyer_notification(titre, message)

if __name__ == "__main__":
    main()
